user/views.py

```diff
@@ -1,3 +1,258 @@
+            users    = User.objects.get(email=email)
+
+            user_data = {
+                'id'               : users.id,
+                'email'            : users.email,
+                'phone'            : users.phone,
+                'name'             : users.name,
+                'profile_image_url': users.profile_image_url
+            }
+
+            if bcrypt.checkpw(password.encode('utf-8'), get_pw) :
+                access_token    = jwt.encode({'user_id' : users.id}, SECRET, algorithm = ALGORITHM).decode('utf-8')
+                return JsonResponse ({"message":"SUCCESS",'authorization':access_token, 'user_data' : user_data },status=200)
+
+            return JsonResponse({"message":"INVALID EMAIL OR PASSWORD"},status=400)
+
+        except KeyError :
+            return JsonResponse({"message":"Key_Error"},status=400)
+
+class KakaoLoginView(View):
+    def post(self, request):
+        try:
+            API_HOST = 'https://kapi.kakao.com/v2/user/me'
+
+            access_token    = request.headers.get('Authorization')
+            print(access_token)
+            request_headers = {'Authorization' : f'Bearer {access_token}'}
+
+            response  = requests.get(API_HOST, headers = request_headers)
+            user_data = json.loads(response.text)['kakao_account']
+
+            if not User.objects.filter(email = user_data['email']).exists():
+                User.objects.create(
+                    email             = user_data['email'],
+                    password          = '',
+                    phone             = '',
+                    name              = user_data['profile']['nickname'],
+                    profile_image_url = user_data['profile']['profile_image_url']
+                )
+
+            users = User.objects.get(email = user_data['email'])
+            user_data = {
+                'id'               : users.id,
+                'email'            : users.email,
+                'phone'            : users.phone,
+                'name'             : users.name,
+                'profile_image_url': users.profile_image_url
+            }
+            winted_token    = jwt.encode({'user_id' : users.id}, SECRET, algorithm = ALGORITHM).decode('utf-8')
+
+            return JsonResponse({'message' : 'SUCCESS', 'authorization': winted_token, 'user_data' : user_data}, status = 200)
+
+        except KeyError as ex:
+            return JsonResponse({'message' : 'KEY_ERROR_' + ex.args[0]}, status = 400)
+        except ValueError as ex:
+            return JsonResponse({'message' : 'VALUE_ERROR_' + ex.args[0]}, status = 400)
+
+            
+class LikeView(View):
+    @token_check
+    def post(self , request):
+        try:
+            data            = json.loads(request.body)
+            company_id      = data["company_id"]
+            user_id         = request.user.id   
+            get_user_id     = User.objects.get(id=user_id).id
```

Comment:
이미 user 인스턴스에서 id 값을 불러오고 있기 때문에 한번 더 User.objects.get()을 사용할 필요가 없습니다.
```suggestion
            user_id = request.user.id
```

Reply:
필수